---
id: 65f9b9710cca621244d3bde1
title: Step 35
challengeType: 20
dashedName: step-35
---

# --description--

Replace `pass` with a `raise` statement that raises a `TypeError` and use the following f-string to provide a custom error message: `f"Addition is not defined between '{self.__class__.__name__}' and '{other.__class__.__name__}'"`.


# --hints--

You should raise a `TypeError` and use the provided string to return a custom error message inside the `if` body.

```js
({
    test: () => assert(runPython(`
      _Node(_code).find_class("R2Vector").find_function("__add__").find_ifs()[0].find_bodies()[0].has_stmt('raise TypeError(f"Addition is not defined between \\'{self.__class__.__name__}\\' and \\'{other.__class__.__name__}\\'")')
    `))
})
```

# --seed--

## --seed-contents--

```py
class R2Vector:
    def __init__(self, *, x, y):
        self.x = x
        self.y = y

    def norm(self):
        return sum(val**2 for val in self.__dict__.values())**0.5

    def __str__(self):
        return str(tuple(self.__getattribute__(i) for i in self.__dict__))

    def __repr__(self):
        arg_list = [f'{key}={val}' for key, val in self.__dict__.items()]
        args = ', '.join(arg_list)
        return f'{self.__class__.__name__}({args})'
--fcc-editable-region--
    def __add__(self, other):
        if not isinstance(other, self.__class__):
            pass
--fcc-editable-region--
class R3Vector(R2Vector):
    def __init__(self, *, x, y, z):
        super().__init__(x=x, y=y)
        self.z = z

v1 = R2Vector(x=2, y=3)
v2 = R3Vector(x=2, y=2, z=3)
print(f'v1 = {v1}')
print(f'v2 = {v2}')
```
